test_communication/CMakeLists.txt

```diff
@@ -70,17 +70,24 @@ if(BUILD_TESTING)
     if(NOT target_path)
       set(target_path "${CMAKE_CURRENT_BINARY_DIR}")
     endif()
-    # testing this with a single message type should be enough
-    ament_add_test(
-      "${target}${target_suffix}"
-      COMMAND "${target_path}/${target}${target_suffix}" Empty
-      TIMEOUT 15
-      GENERATE_RESULT_FOR_RETURN_CODE_ZERO
-      ENV RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
-    set_tests_properties(
-      ${target}${target_suffix}
-      PROPERTIES REQUIRED_FILES "${target_path}/${target}${target_suffix}"
-    )
+    # adding test for each message type
+    foreach(message_file ${message_files})
+      get_filename_component(TEST_MESSAGE_TYPE "${message_file}" NAME_WE)
+      # TODO(dirk-thomas) publishing DynamicArrayPrimitives in OpenSplice is buggy
+      # https://github.com/ros2/rclcpp/issues/219
+      if(NOT "${TEST_MESSAGE_TYPE}" STREQUAL "DynamicArrayPrimitives" OR NOT "${rmw_implementation1}" STREQUAL "rmw_opensplice_cpp")
```

Comment:
Since we are using CMake 3.5 now please avoid the quotes and explicit variable expansion if the variable is know to exist:

    if(NOT TEST_MESSAGE_TYPE STREQUAL "DynamicArrayPrimitives" OR NOT rmw_implementation1 STREQUAL "rmw_opensplice_cpp")